cryptographic_estimators/MAYOEstimator/MAYOAlgorithms/claw_finding.py

```diff
@@ -0,0 +1,151 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..mayo_algorithm import MAYOAlgorithm
+from ..mayo_problem import MAYOProblem
+from ...MQEstimator.mq_estimator import MQEstimator
+from ...MQEstimator.MQAlgorithms.lokshtanov import Lokshtanov
+from ...base_constants import BASE_FORGERY_ATTACK
+from ...base_algorithm import optimal_parameter
+from math import log2, e
+
+
+class ClawFinding(MAYOAlgorithm):
+    """
+    Construct an instance of ClawFinding estimator
+
+    Claw finding attack is a general attack which works against any signature which 
+    follows the hash-and-sign paradigm. 
+
+    INPUT:
+
+    - ``problem`` -- MAYOProblem object including all necessary parameters
+    - ``w`` -- linear algebra constant (default: Obtained from MAYOAlgorithm)
```

Comment:
The default should be the same as in the MAYOEstimator which currently is stating that the default value of `w` is `2`, this value should be 2.81 by default in the MAYOEstimator